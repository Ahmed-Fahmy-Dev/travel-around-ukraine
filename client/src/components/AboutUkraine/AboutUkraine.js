/* eslint-disable react/jsx-props-no-spreading */
import React from 'react'
import {Box, Typography, styled, keyframes  } from '@mui/material';
// import { useInView } from 'react-intersection-observer';

const UkraineTextInfo = styled((props) => <Typography  {...props} />)(({ theme }) => ({
    [theme.breakpoints.up('tablet')]: {
        padding: '10px',
    },

    [theme.breakpoints.up('laptop')]: {
        padding: '20px'
    },
}));

const AboutUkraineContainer = styled((props) => <Box {...props} />)(({ theme }) => ({
    backgroundColor: '#EDEDED',
    borderRadius: '20px',
    zIndex: 1,
    transition: '2s',
    position: 'absolute',
    top: '12%',
    left: '40%',
    width: '200px',
    marginRight: '30px',
    padding: '20px',
   
    '&:hover':{
        backgroundColor: theme.palette.primary.main,
        animation: `
        -webkitTransform: translateX(-60px);
        transform: translateX(-60px);`,
        boxShadow: '1px 1px #ededed, 2px 2px #ededed, 3px 3px #ededed, 4px 4px #ededed, 5px 5px #ededed, 6px 6px #ededed,  7px 7px #ededed',
    },

    '&:hover h2':{
       color: theme.palette.secondary.main
    },

    [theme.breakpoints.up('tablet')]: {
       top:'15%',
       width: '400px',
       left: '52%',
       padding: '25px',
    },

    [theme.breakpoints.up('laptop')]: {
        top:'20%',
        left: '55%',
        width: '600px',
        padding: '30px',
    },
}));

const fadeIn = keyframes`
0% {
    opacity: 0;
  }
  100% {
    opacity: 1;
  }
`;

// ../../img/ukraine.jpeg

const UkraineBackground = styled((props) => <Box {...props} />)(({ theme }) => ({
    height: 800,
    backgroundImage: `url(${"../../assets/imgAboutUkraineSection/ukraine.jpeg"})`,
    backgroundSize: 'cover',
    backgroundPosition: 'center',
    borderRadius: '20px',
    objectFit: 'cover',
    marginRight:'124px',
    animation: `${fadeIn} 3s;`,
    
    [theme.breakpoints.up('tablet')]: {
        marginRight:'264px',
    },

    [theme.breakpoints.up('laptop')]: {
        marginRight:'369px',
    }, 
}));


function AboutUkraine() {
    // const { ref, inView } = useInView({
    //     threshold: 0
    //   });
    return (
        <UkraineBackground sx={{ position: 'relative' }} >
                <AboutUkraineContainer >
                    <Typography variant='h2' >What is Ukraine?</Typography>
                        <UkraineTextInfo >You are going to explore the largest country in Europe – a picturesque place where innovative creativity meets ancient traditions, and wonderful legends come into reality.
                            Upon the road you will come to know lots of brave and kind-hearted people.
                            They are fighting for freedom and dignity.
                            They are searching for peace and prosperity.
                            They are transforming Ukraine into a modern country, unique business hub and new all-season tourist destination.
                        </UkraineTextInfo>
                        <UkraineTextInfo >
                            Now get closer. The time has come to unlock the great and still less than fully realized potential of Ukraine.
                            It is our time.
                            And it starts right NOW.
                            So, you’ve got the answer to the question ‘What is Ukraine?’… Welcome, and don’t forget to take a big backpack for some great memories with you!
                        </UkraineTextInfo>
                </AboutUkraineContainer>
        </UkraineBackground>
    );
}

export default AboutUkraine;
