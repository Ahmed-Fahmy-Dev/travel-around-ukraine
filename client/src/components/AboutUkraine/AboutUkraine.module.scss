.flex {
  display: flex;
  position: relative;
  margin-right: 300px;
}

.flex_img {
  display: flex;
  margin-right: 200px;

  animation: fadeIn 6s;
  // -webkit-animation: fadeIn 6s;
  // -moz-animation: fadeIn 6s;
  // -o-animation: fadeIn 6s;
  // -ms-animation: fadeIn 6s;
}

@keyframes fadeIn {
  0% {
    opacity: 0;
  }

  100% {
    opacity: 1;
  }
}

@-moz-keyframes fadeIn {
  0% {
    opacity: 0;
  }

  100% {
    opacity: 1;
  }
}

@-webkit-keyframes fadeIn {
  0% {
    opacity: 0;
  }

  100% {
    opacity: 1;
  }
}

@-o-keyframes fadeIn {
  0% {
    opacity: 0;
  }

  100% {
    opacity: 1;
  }
}

@-ms-keyframes fadeIn {
  0% {
    opacity: 0;
  }

  100% {
    opacity: 1;
  }
}


.what_is_ukraine {
  font-family: 'Montserrat';
  font-style: normal;
  font-weight: 700;
  font-size: 24px;
  line-height: 29px;
  color: #0499DD;
}


.styles.what_is_ukraine_description {
  font-family: 'Montserrat';
  font-style: normal;
  font-weight: 500;
  font-size: 14px;
  line-height: 17px;
  color: #282828;
}

.info:hover {
  background-color: #0499DD;
  transition: 3s;
  box-shadow:
    1px 1px #EDEDED,
    2px 2px #EDEDED,
    3px 3px #EDEDED,
    4px 4px #EDEDED,
    5px 5px #EDEDED,
    6px 6px #EDEDED,
    7px 7px #EDEDED;
  -webkit-transform: translateX(-60px);
  transform: translateX(-60px);
}

.info:hover .what_is_ukraine {
  color: #EDEDED;
}


.what_is_ukraine_img {
  height: 704px;
  width: 100%;
  border-radius: 20px;
  object-fit: cover;
}


.info {
  box-sizing: border-box;
  padding: 20px;
  background-color: #EDEDED;
  border-radius: 10px;
  z-index: 1;
  position: absolute;
  right: 24px;
  top: 20%;
  margin: 0 24px;
}




@media (min-width:550px) {
  .info {
    top: 28%;
    padding: 30px;
    width: 450px;
  }

}

@media (min-width:610px) {
  .info {
    top: 35%;
    padding: 30px;
    width: 494px;
  }


}

@media (min-width:1024px) {
  .info {
    top: 30%;
    padding: 40px;
  }

}