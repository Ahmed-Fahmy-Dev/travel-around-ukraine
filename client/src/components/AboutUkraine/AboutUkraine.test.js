/* eslint-disable react/react-in-jsx-scope */
/* eslint-disable no-undef */
import {  render,  } from '@testing-library/react';
import AboutUkraine from './AboutUkraine';

describe('AboutUkraine snapshot testing', () => {
test('should render AboutUkraine Component', () => { 
   const {asFragment} =  render(<AboutUkraine/>);
   expect(asFragment()).toMatchSnapshot();
 })
});