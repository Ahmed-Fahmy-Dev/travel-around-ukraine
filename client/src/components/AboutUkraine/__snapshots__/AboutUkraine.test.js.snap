// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutUkraine snapshot testing should render AboutUkraine Component 1`] = `
<DocumentFragment>
  <section
    style="opacity: 0; transform: translateX(-100px) translateZ(0);"
  >
    <div
      class="MuiBox-root css-qbb5fe"
    >
      <div
        class="MuiBox-root css-y2ilk6"
      >
        <h2
          class="MuiTypography-root MuiTypography-h2 css-1sra7t5-MuiTypography-root"
        >
          What is Ukraine?
        </h2>
        <p
          class="MuiTypography-root MuiTypography-body1 css-66u9a6-MuiTypography-root"
        >
          You are going to explore the largest country in Europe – a picturesque place where innovative creativity meets ancient traditions, and wonderful legends come into reality. Upon the road you will come to know lots of brave and kind-hearted people. They are fighting for freedom and dignity. They are searching for peace and prosperity. They are transforming Ukraine into a modern country, unique business hub and new all-season tourist destination.
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-66u9a6-MuiTypography-root"
        >
          Now get closer. The time has come to unlock the great and still less than fully realized potential of Ukraine. It is our time. And it starts right NOW. So, you’ve got the answer to the question ‘What is Ukraine?’… Welcome, and don’t forget to take a big backpack for some great memories with you!
        </p>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
